sklearn/linear_model/coordinate_descent.py

```diff
@@ -99,9 +99,12 @@ def _alpha_grid(X, y, Xy=None, l1_ratio=1.0, fit_intercept=True,
             Xy /= X_std[:, np.newaxis]
     alpha_max = (np.sqrt(np.sum(Xy ** 2, axis=1)).max() /
                  (n_samples * l1_ratio))
-    alphas = np.logspace(np.log10(alpha_max * eps), np.log10(alpha_max),
-                         num=n_alphas)[::-1]
-    return alphas
+
+    if alpha_max == 0:
+        return np.zeros(n_alphas)
+
+    return np.logspace(np.log10(alpha_max * eps), np.log10(alpha_max),
+                       num=n_alphas)[::-1]
```

Comment:
this produces warnings in the test suite.

coordinate_descent.py:421: UserWarning: Coordinate descent with alpha=0 may lead to unexpected results and is discouraged.
  tol, rng, random, positive)